File: server/src/main/java/dev/oscarrojas/drawandguess/io/Action.java
package dev.oscarrojas.drawandguess.io;

import dev.oscarrojas.drawandguess.dto.CreateLobby;

public enum Action {
    CREATE_LOBBY(CreateLobby.class);

    public final Class<?> payloadType;

    Action(Class<?> payloadType) {
        this.payloadType = payloadType;
    }
}

File: server/src/main/typescript/Action.ts
export enum Action {
  CREATE_LOBBY = "CREATE_LOBBY",
}

export function getAction(value: string): Action {
  if (value === Action.CREATE_LOBBY) {
    return Action.CREATE_LOBBY;
  } else {
    throw new Error("Unknown action value: " + value);
  }
}
